aiohttp/multipart.py

```diff
@@ -310,13 +311,14 @@ def _read_chunk_from_stream(self, size):
             self._prev_chunk = yield from self._content.read(size)
 
         chunk = yield from self._content.read(size)
-
+        self._content_eof += int(self._content.at_eof())
+        assert self._content_eof < 3, "Reading after EOF"
```

Comment:
You actually need this check only for new line reading: if it doesn't starts with like a boundary, no need to buffer anything. Could be controlled with a simple flag.